packages/sanity/src/core/bundles/hooks/usePerspective.tsx

```diff
@@ -15,33 +19,45 @@ export interface PerspectiveValue {
 }
 
 /**
+ * TODO: Improve distinction between global and pane perspectives.
+ *
  * @internal
  */
 export function usePerspective(): PerspectiveValue {
   const router = useRouter()
   const {data: bundles} = useBundles()
+  const perspective = useNearestPerspective()
 
+  // TODO: Should it be possible to set the perspective within a pane, rather than globally?
   const setPerspective = (slug: string | undefined) => {
     if (slug === 'drafts') {
       router.navigateStickyParam('perspective', '')
     } else {
       router.navigateStickyParam('perspective', `bundle.${slug}`)
     }
   }
+
   const selectedBundle =
-    router.stickyParams?.perspective && bundles
+    perspective && bundles
       ? bundles.find((bundle: Partial<BundleDocument>) => {
-          return (
-            `bundle.${bundle.slug}`.toLocaleLowerCase() ===
-            router.stickyParams.perspective?.toLocaleLowerCase()
-          )
+          return `bundle.${bundle.slug}`.toLocaleLowerCase() === perspective?.toLocaleLowerCase()
         })
       : LATEST
 
+  // TODO: Improve naming; this may not be global.
   const currentGlobalBundle = selectedBundle || LATEST
 
   return {
     setPerspective,
     currentGlobalBundle,
   }
 }
+
+/**
+ * @internal
+ */
+export function useNearestPerspective(): string | undefined {
+  const router = useRouter()
+  const paneRouter = usePaneRouter()
```

Comment:
Given this `usePerspective` hook is part of core, I think we shouldn't use here the `paneRouter`

Reply:
That's a good point. Maybe `useNearestPerspective` should move. I need to think about the best way to proceed here... if you have any thoughts, let me know 🙏.